File: src/main/java/com/github/caijh/commons/util/process/Command.java
package com.github.caijh.commons.util.process;

public interface Command {

    String[] getCmdArray();

    default ProcessResult exec() {
        ProcessResult result = new ProcessResult();
        try {
            String[] cmdArray = getCmdArray();
            Process process = Runtime.getRuntime().exec(cmdArray);
            process.waitFor();
            int exitValue = process.exitValue();
            String consoleString = ProcessUtils.readConsoleString(process);
            result.setExitValue(exitValue);
            result.setConsoleString(consoleString);
        } catch (Exception e) {
            throw new ProcessException("call process fail", e);
        }
        return result;
    }

}

File: src/main/kotlin/com/github/caijh/commons/util/process/Command.kt
package com.github.caijh.commons.util.process

interface Command {
    val cmdName: String?
    val cmdArray: Array<String>
    fun exec(): ProcessResult? {
        val result = ProcessResult()
        try {
            val cmdArray = cmdArray
            val process = Runtime.getRuntime().exec(cmdArray)
            process.waitFor()
            val exitValue = process.exitValue()
            val consoleString = ProcessUtils.readConsoleString(process)
            result.exitValue = exitValue
            result.consoleString = consoleString
        } catch (e: Exception) {
            throw ProcessException("call process fail", e)
        }
        return result
    }
}
